docs/discussion/selection-api.md

```diff
@@ -0,0 +1,378 @@
+
+We evaluated the current selection features available in ipyaladin, and
+prototyped programmatic extensions to support:
+
+- selecting a list of sources
+  [[PR](https://github.com/pcuste1/ipyaladin/pull/2)]
+
+- selecting sources within a region
+  [[PR](https://github.com/pcuste1/ipyaladin/pull/1)].
+
+These proofs-of-concept worked by reaching into internal components of
+Aladin Lite. However, a more robust and maintainable solution requires
+official API support.
+
+This document outlines:
+
+- our design principles
+
+- feature requirements
+
+- API specifications for both Aladin Lite and ipyaladin
+
+- several motivating use cases.
+
+## 1.1. Design Principles
+
+#### Standard Interface Pattern
+
+For each feature exposed by **aladin-lite** that we intend to support,
+the interface should consistently offer the following APIs:
+
+- **Listener**: Allows developers to attach callbacks to respond to
+  relevant events.
+
+- **Get**: Retrieves the current state or value of the feature.
+
+- **Set (Select)**: Updates the feature\'s state programmatically. 
+
+- **Misc:** Features that support the other 3,
+
+This pattern ensures clarity, predictability, and ease of use across all
+supported functionality.
+
+#### Separation of Concerns
+
+It is essential to clearly distinguish between the roles of the
+ipyaladin and aladin-lite APIs. While many capabilities in ipyaladin
+serve as direct proxies for aladin-lite features, this is not always the
+case. ipyaladin is a higher-level interface that may combine
+multiple aladin-lite operations to deliver more integrated or complex
+functionality.
+
+For example, although aladin-lite may expose individual APIs for
+selecting catalog objects and footprints, ipyaladin might provide a
+unified API that coordinates both actions under a single interface call.
+
+#### Consistent Data Formats
+
+To streamline development and enhance user experience, all ipyaladin
+interfaces are expected to rely on Astropy-defined object formats (e.g.,
+[Table](https://docs.astropy.org/en/stable/table/index.html),
+[Region](https://astropy-regions.readthedocs.io/en/stable/)).
+This provides a familiar and well-documented foundation for users
+working within the scientific Python ecosystem.
+
+#### User-Defined Catalogs
+
+All selection interfaces are designed with the assumption that catalogs
+are loaded into aladin-lite programmatically by the user within their
+Jupyter notebook. Catalogs added directly via the widget interface are
+not considered within the scope of these APIs, ensuring greater control
+and consistency for programmatic workflows.
+
+## 1.2. Feature Requirements
+
+#### Selection Requirements
+
+As a notebook user/scientist I want to\...
+
+- slect a source within a catalog using a programmatic API
+
+- select a list of sources within all loaded catalogs using a
+  programmatic API
+
+- select all sources within a defined region using a programmatic API
+
+- select a footprint using a programmatic API 
```

Comment:
As a general comment for all the footprint API things here, we need to support selection of multiple footprints.  Whether we move more towards a model of unifying the handling of footprints and catalog objects is TBD, but conceptually I think the handling should be fairly similar.